lib/model/message_list.dart

```diff
@@ -448,25 +448,27 @@ mixin _MessageSequence {
     required MessageListMessageBaseItem Function(bool canShareSender) buildItem,
   }) {
     final bool canShareSender;
-    if (
-      prevMessage == null
-      || oneMessagePerBlock
-      || !haveSameRecipient(prevMessage, message)
-    ) {
+    if (prevMessage == null) {
       items.add(MessageListRecipientHeaderItem(message));
       canShareSender = false;
     } else {
       assert(items.last is MessageListMessageBaseItem);
       final prevMessageItem = items.last as MessageListMessageBaseItem;
       assert(identical(prevMessageItem.message, prevMessage));
-      assert(prevMessageItem.isLastInBlock);
-      prevMessageItem.isLastInBlock = false;
+      assert(prevMessageItem.isLastInBlock); // though this may mutate; see below
 
-      if (!messagesSameDay(prevMessageItem.message, message)) {
-        items.add(MessageListDateSeparatorItem(message));
+      if (oneMessagePerBlock || !haveSameRecipient(prevMessage, message)) {
+        items.add(MessageListRecipientHeaderItem(message));
         canShareSender = false;
       } else {
-        canShareSender = prevMessageItem.message.senderId == message.senderId;
+        prevMessageItem.isLastInBlock = false;
```

Comment:
Yeah, I think the refactors in the middle two commits are no longer helpful. The assertions are really there to support the `prevMessageItem` local, and then the last one is for checking the precondition of the assignment right after it:
```dart
      assert(prevMessageItem.isLastInBlock);
      prevMessageItem.isLastInBlock = false;
```

So separating them from that line, and from the uses of `prevMessageItem` in general, isn't an improvement.